src/main/java/de/tum/in/www1/artemis/service/export/CourseExamExportService.java

```diff
@@ -410,30 +411,24 @@ private List<Path> exportExercises(String notificationTopic, Set<Exercise> exerc
-                    exportedExercises.add(fileUploadExerciseWithSubmissionsExportService.exportFileUploadExerciseWithSubmissions(exercise, submissionsExportOptions,
-                            exerciseExportDir, exportErrors, reportData));
-                }
-                else if (exercise instanceof TextExercise) {
-                    exportedExercises.add(textExerciseWithSubmissionsExportService.exportTextExerciseWithSubmissions(exercise, submissionsExportOptions, exerciseExportDir,
-                            exportErrors, reportData));
-                }
-                else if (exercise instanceof ModelingExercise) {
-                    exportedExercises.add(modelingExerciseWithSubmissionsExportService.exportModelingExerciseWithSubmissions(exercise, submissionsExportOptions, exerciseExportDir,
-                            exportErrors, reportData));
-                }
-                else if (exercise instanceof QuizExercise quizExercise) {
-                    exportedExercises.add(quizExerciseWithSubmissionsExportService.exportExerciseWithSubmissions(quizExercise, exerciseExportDir, exportErrors, reportData));
-
-                }
-                else {
-                    // Exercise is not supported so skip
+                switch (exercise) {
+                    case ProgrammingExercise programmingExercise ->
+                        // Download the repositories' template, solution, tests and students' repositories
+                        programmingExerciseExportService.exportProgrammingExerciseForArchival(programmingExercise, exportErrors, Optional.of(exerciseExportDir), reportData)
+                                .ifPresent(exportedExercises::add);
+
+                    // Export the other exercises types
+                    case FileUploadExercise fileUploadExercise -> exportedExercises.add(fileUploadExerciseWithSubmissionsExportService
+                            .exportFileUploadExerciseWithSubmissions(fileUploadExercise, submissionsExportOptions, exerciseExportDir, exportErrors, reportData));
+                    case TextExercise textExercise -> exportedExercises.add(textExerciseWithSubmissionsExportService.exportTextExerciseWithSubmissions(textExercise,
+                            submissionsExportOptions, exerciseExportDir, exportErrors, reportData));
+                    case ModelingExercise modelingExercise -> exportedExercises.add(modelingExerciseWithSubmissionsExportService
+                            .exportModelingExerciseWithSubmissions(modelingExercise, submissionsExportOptions, exerciseExportDir, exportErrors, reportData));
+                    case QuizExercise quizExercise ->
+                        exportedExercises.add(quizExerciseWithSubmissionsExportService.exportExerciseWithSubmissions(quizExercise, exerciseExportDir, exportErrors, reportData));
+                    default -> {
+                        // Exercise is not supported so skip
```

Comment:
Maybe throw an error that the user notices that an exercise was not handled correctly?

Reply:
I'd prefer to leave it like that:
If you have an erroneous exercise, the export breaks. And you don't have a chance to export anything unless you - at worst delete the exercise.

The best solution would be to write an export summary that gets displayed to the user; however, that's a completely new feature implementation.

Reply:
I added an error to the error list